patch/cisco-npu-disable-other-bars.patch

```diff
@@ -0,0 +1,71 @@
+From d9ddce0a1c4699cf6c2c139da907a09cc1255178 Mon Sep 17 00:00:00 2001
+From: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+Date: Mon, 15 Nov 2021 23:22:15 -0800
+Subject: [PATCH] NPU disable unused PCI BAR's
+
+For Cisco ASIC only BAR0 is valid. Not disabling other BAR's
+was resulting in pci_enable_device function failure in P0
+Pacific ASIC's. Further debugging and consultion with Hardware
+team, issue seems to be related to only P0 version of ASIC and
+workaround suggested is to disable unused PCI BAR.
+
+Signed-off-by: Madhava Reddy Siddareddygari <msiddare@cisco.com>
+---
+ drivers/pci/quirks.c | 44 ++++++++++++++++++++++++++++++++++++++++++++
+ 1 file changed, 44 insertions(+)
+
+diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
+index 16fb3d771..a48efa6fe 100644
+--- a/drivers/pci/quirks.c
++++ b/drivers/pci/quirks.c
+@@ -5686,3 +5686,47 @@ static void apex_pci_fixup_class(struct pci_dev *pdev)
+ }
+ DECLARE_PCI_FIXUP_CLASS_HEADER(0x1ac1, 0x089a,
+ 			       PCI_CLASS_NOT_DEFINED, 8, apex_pci_fixup_class);
++
++#define PCI_DEVICE_ID_LEABA_PACIFIC    0xabcd
++#define PCI_DEVICE_ID_LEABA_GIBRALTAR  0xa001
++#define PCI_DEVICE_ID_LEABA_GRAPHENE   0xa003
++#define PCI_DEVICE_ID_LEABA_PALLADIUM  0xa004
++#define PCI_DEVICE_ID_LEABA_ARGON      0xa005
++#define PCI_DEVICE_ID_LEABA_KRYPTON    0xa006
++
++/*
++ * For Pacific A0, only BAR 0 is valid
++ */
++static void silicon_one_fixup(struct pci_dev *dev)
++{
++	int i;
++	struct resource *r;
++
++	for (i = 1; i <= PCI_ROM_RESOURCE; i++) {
++		r = &dev->resource[i];
++		if (!r->start && !r->end && !r->flags)
++			continue;
++
++		dev_info(&dev->dev, "Cisco Silicon One BAR %d %pR fixed up\n", i, r);
```

Comment:
Can you change `dev_info` to `pci_info`, for logging consistency with the rest of the file?

Reply:
Updated patch with review comments